public/js/clients/firefox/commands.js

```diff
@@ -54,22 +54,26 @@ function setBreakpoint(location, condition, noSliding) {
     column: location.column,
     condition,
     noSliding
-  }).then(([res, bpClient]) => {
-    bpClients[bpClient.actor] = bpClient;
-
-    // Firefox only returns `actualLocation` if it actually changed,
-    // but we want it always to exist. Format `actualLocation` if it
-    // exists, otherwise use `location`.
-    const actualLocation = res.actualLocation ? {
-      sourceId: res.actualLocation.source.actor,
-      line: res.actualLocation.line,
-      column: res.actualLocation.column
-    } : location;
-
-    return BreakpointResult({
-      id: bpClient.actor,
-      actualLocation: Location(actualLocation)
-    });
+  }).then((res) => onNewBreakpoint(location, res))
+}
+
+function onNewBreakpoint(location, res) {
+  const bpClient = res[1];
+  let actualLocation = res[0].actualLocation;
+  bpClients[bpClient.actor] = bpClient;
```

Comment:
also, setCondition does not return `actualCondition`, but that is likely okay because you're just editing the bp.